lib/queue.rb

```diff
@@ -1,31 +1,61 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(50)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      @front = 0
+      @back = 1
+      @store[@front] = element
+    elsif @front == @back
+      raise ArgumentError
+    else
+      @store[@back] = element
+      @back = (@back + 1 ) % @store.length
+    end
   end
 
-  def dequeue
-    raise NotImplementedError, "Not yet implemented"
+  def dequeue #remove and return from front
+    if @front == -1
+      raise ArgumentError
+    else
+      element = @store[@front]
+      @front = (@front + 1) % @store.length
+    end
+
+    if @front == @back
+      @front = @back = -1
+    end
+
+    return element
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    @back - @front
```

Comment:
This works when front < back.  What about when back has wrapped around the buffer.